Ignore PRs by specified usernames in the generated changelist

agriyakhetarpal opened the pull request:
> [!NOTE]
> This PR is a draft, and its description is a stub. I'll update it soon! 

Closes #83

TODO:
- [x] add tests
- [x] update README
- [x] do a sanity check once —> @stefanv did this in https://github.com/scientific-python/changelist/pull/85#issuecomment-3610623422; thanks :)

## Release note

For maintainers and optionally contributors, please refer to the [instructions](https://scikit-image.org/docs/stable/development/contribute.html#documenting-changes) on how to document this PR for the release notes.

```release-note
By default, hide pull requests that are authored by users / bots given in
the config field `ignored_user_logins. This behavior can be overridden by
explicitly specifying the new config field `ignore_prs_by_username`.
```

agriyakhetarpal committed:
add `ignored_user_logins` param to `ChangeNote.from_pull_requests`

stefanv commented:
I'd really appreciate having this feature; trying to release yaml2ics now, and the changelog is filled with "Bump x to version y"

stefanv commented:
I can confirm that this PR did the trick :)

lagru committed:
Add `ignore_prs_by_username` field with include option

This allows users to configure ignored contributors and which PRs are
ingnored by users independently. At the same time most users for which
these two fields may look the same don't have to repeat themselves.

lagru committed:
Test new `ignore_prs_by_username` option

lagru committed:
Use `Optional` instead of `| None` to support Python 3.9

agriyakhetarpal commented:
Ah, I was going to push changes just now :( thanks for taking it further up anyway, @lagru!

lagru committed:
Fix erroneous refactor

lagru committed:
Apply dereferencing last when adding defaults

Dereferencing can only be done once both fields have been loaded which
may happen only at this stage when adding defaults.

lagru commented:
@agriyakhetarpal sorry for my bad timing! I hope I didn't waste too much of your time here and that you didn't feel like I stepped on your toes. :pray: Since there was some demand for it, I thought it quickest to finish this up myself. I'll check-in with you next time!

lagru commented:
@stefanv I've added a new config field `ignore_prs_by_username` that defaults to `ignored_user_logins`. I confirmed that hides the "Bump" PRs in https://github.com/scientific-python/lazy-loader/releases/tag/v0.4 when regenerating the notes with
```
changelist scientific-python/lazy-loader v0.3 v0.4
```

I'll wait a bit in case @stefanv wants to test this again or in case @agriyakhetarpal has feedback or wants to push more changes.

stefanv commented:
Thanks so much, @lagru and @agriyakhetarpal!
